### PR TITLE
Change `Share` icon and improve layout of item header actions

### LINKED ISSUE #550: Update "Share" icon
Use an icon that shows people for the "Share" menu.
Use the Sharing icon for the "Publish" menu.

### PR DESCRIPTION
**Pull request by spaenleh**
In this PR I propose to change the `Share` icon to better represent that this menu is related to sharing with people.

Previously:
previously-share-icon

Proposed change:
proposed-share-icon

And the link box was styled with the `text-overflow: ellipsis` to better show the link could not be displayed fully:
link box

IconButtons were also all wrapped in `span`s to ensure they stay round.

TODO:
- [x] create a version of `@graasp/ui` with the changes from https://github.com/graasp/graasp-ui/pull/245 and https://github.com/graasp/graasp-ui/commit/88771117906c8fdc3c228ff8a464b14e94fbe148

Change `Favorite Items` icon from a heart to a star to be in ligne with the `Add to Favorites` icon in item actions.
Capture d’écran 2023-02-21 à 22 58 44

closes #550

### CODE REVIEW

File: src/components/item/sharing/SharingLink.tsx [hunk 1]
```diff
@@ -88,10 +85,20 @@ const SharingLink: FC<Props> = ({ itemId }) => {
...
+            payload: { message: translateBuilder('Link copied to clipboard') },
+          });
         },
         onError: () => {
-          notifier({ type: COPY_ITEM_LINK_TO_CLIPBOARD.FAILURE, payload: {} });
+          notifier({
+            type: COPY_ITEM_LINK_TO_CLIPBOARD.FAILURE,
+            payload: {
+              message: translateBuilder(
+                'There was an error copying the link to the clipboard',
```
Comment (pyphilia) [inline src/components/item/sharing/SharingLink.tsx:98]: Should we add these strings in translations?

**Review by pyphilia — APPROVED** [review]
Looks good! I feel like the new share button is smaller than the other ones?

**Comment by spaenleh** [comment]
Thank you for the review @pyphilia. I don't think it is smaller but i might appear a bit smaller due to the design:
Capture d’écran 2023-02-22 à 10 59 59
Capture d’écran 2023-02-22 à 10 59 51
Capture d’écran 2023-02-22 à 10 59 42

Do you think it is an issue ? I don't think we should "compensate" visually by making the icon bigger. 
Do you see an icon in [the MUI Material Icons list](https://mui.com/material-ui/material-icons/) that would be better suite ?
Maybe [the `Group` icon](https://mui.com/material-ui/material-icons/?query=group&selected=Group) which displays 2 people instead of 3 ?

File: src/components/item/sharing/SharingLink.tsx (hunk 1 above)
Comment (spaenleh) [reply src/components/item/sharing/SharingLink.tsx:98]: @pyphilia see https://github.com/graasp/graasp-translations/pull/110

**Pull request merged**